### PR TITLE
[Chef-17] 1 of X - Updating Ruby versions

### PR DESCRIPTION
**Pull request by johnmccrae**
## Description
Chef-17 Ruby versions need to be updated to reflect dropping Ruby 2.6 support and being consistent about Ruby 2.7 & 3.0 support

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: knife/knife.gemspec [hunk 1]
```diff
@@ -13,7 +13,7 @@ Gem::Specification.new do |s|
   s.email = "adam@chef.io" # These seem a bit out of date, and this address probably doesn't go anywhere anymore?
   s.homepage = "https://www.chef.io"
 
-  s.required_ruby_version = ">= 2.7.0"
+  s.required_ruby_version = ">= 3.0.0"
```
Comment (neha-p6) [inline knife/knife.gemspec:16]: @johnmccrae knife is already at >= 2.7.0, so this change should not be needed right? This is breaking change

File: .expeditor/scripts/bk_win_functional.ps1 [hunk 2]
```diff
@@ -20,10 +45,10 @@ Write-Output "--- configure winrm"
 winrm quickconfig -q
 
 Write-Output "--- bundle install"
-bundle config set --local without 'omnibus_package'
+bundle config set --local without "omnibus_package"
```
Comment (neha-p6) [inline .expeditor/scripts/bk_win_functional.ps1:48]: why we had to do this?

**Review by neha-p6 — CHANGES_REQUESTED** [review]

File: knife/knife.gemspec (hunk 1 above)
Comment (ramereth) [reply knife/knife.gemspec:16]: We should not be bumping this on Chef 17

Comment (johnmccrae) [reply knife/knife.gemspec:16]: That's a good point. I was afraid that if we kept it at 2.7 there might be test problems. I'll roll it back.

File: .expeditor/scripts/bk_win_functional.ps1 (hunk 2 above)
Comment (johnmccrae) [reply .expeditor/scripts/bk_win_functional.ps1:48]: I got used to getting linting errors about single vs double quotes and updated this while I was checking things out.

**Pull request merged**